source/leuven/credits.rst

```diff
@@ -48,52 +44,123 @@ Please contact your VSC coordinator/contact or your :ref:`local support staff
 Job cost calculation
 ~~~~~~~~~~~~~~~~~~~~
 
-The cost of a job depends on the resources it consumes. Generally
-speaking, one credit buys the user one hour of walltime on one reference
-node. The resources that are taken into account to charge for a job are
-the walltime it consumed, and the number and type of compute nodes it
-ran on. The following formula is used:
+On Tier-2 clusters, we use Slurm for accounting purposes (on top of resource and 
+job management).
+See :ref:`Slurm accounting <accounting_leuven>` page for additional information.
+In Slurm terminology, the cost of a job depends on the trackable resources (TRES)
+it consumes. Two distinct TRES are the number of CPU cores and GPU devices. 
+Different types of CPU and GPU nodes are given different weights 
+(``TRESBillingWeights``) which you can retrieve as follows for e.g. wICE::
+
+   scontrol show partitions -M wice
+
+As an example, for a GPU node, the weights are configured as::
+
+   TRESBillingWeights=CPU=2.546296296,GRES/gpu:a100-sxm4-80gb=141.6666667,GRES/shard:a100-sxm4-80gb=141.6666667
+
+**CPU-only jobs**
 
-0.000278 \* *nodes* \* *walltime* \* *nodetype*
+The following formula applies::
+
+   CPU TRESBillingWeights * num_cores * walltime
 
 Where
 
--  *nodes* is the number of compute nodes the job ran on;
--  *walltime* the effective duration of the job, expressed in seconds;
--  *nodetype* is the factor representing the node type's performance as
-   listed in the table below.
+- ``CPU TRESBillingWeights`` is the applied weight for CPU resources (see above)
+- ``num_cores`` is the *effective* number of cores used for the job
+- ``walltime`` is the number of minutes that the job ran
+
+**GPU jobs**
+
+The following formula applies::
+
+   (CPU TRESBillingWeights * num_cores + GPU TRESBillingWeights * num_gpus) * walltime
+   
+Where
 
-The Tier-2 cluster has several types of compute nodes, none of which
-is actually a reference node. The values for the different types are
-listed at the following links:
+- ``CPU TRESBillingWeights`` is the applied weight for CPU resources (see above)
+- ``GPU TRESBillingWeights`` is the applied weight for GPU resources (see above)
+- ``num_cores`` is the *effective* number of cores used for the job
+- ``num_gpus`` is the number of GPUs requested for the job
+- ``walltime`` is the number of minutes that the job ran
 
-- :ref:`Genius <running jobs on genius>`
+.. note::
+
+    *Effective* number of cores is not necessarily equal to what the user requests.
+    E.g. if a job requests a single core/task, but the full memory of a node on wICE,
+    then one node is blocked for such a job. Consequently, the effective number of cores
+    will be 72, instead of 1.
+
+    Similarly, on a node with 4 GPUs, if a user requests to use a single GPU, one-fourth
+    of cores will be assigned to the job. So, e.g. on wICE ``num_cores`` will be 18.
```

Comment:
This stil needs to be removed

Reply:
frankly, everytime I tried to test this on wICE, the gpu nodes were busy. But, judging from `slurm.conf` no cpu-to-gpu proportion is set. Also on Slurm documentation, I hardly found a possibility to enforce this. there is only `DefCpuPerGPU` (https://slurm.schedmd.com/slurm.conf.html#OPT_DefCpuPerGPU) which only sets a default value, in case users do not specify it explicitly.

So, I believe we need to enforce this via the submit filter, same as now with PBS on Genius.

So, I will remove that paragraph now.